README.md
![alt XOOPS CMS](https://xoops.org/images/logoXoops4GithubRepository.png)
## SmartClone module for  [XOOPS CMS 2.5.8+](https://xoops.org)
[![XOOPS CMS Module](https://img.shields.io/badge/XOOPS%20CMS-Module-blue.svg)](https://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](LICENSE)

**SmartClone** for [XOOPS CMS](https://xoops.org) is a module that will clone with 2 clicks any module that is developed as clonable! It works with a plugin system so any module developer can create a plugin for its module and allow SmartClone to painlessly clone it!

Current and upcoming "next generation" versions of XOOPS CMS are being crafted on GitHub at: https://github.com/XOOPS


docs/changelog.txt
<h5>1.10 Beta 3  [2017-07-20]</h5>
<hr>

- XOOPS 2.5.9, PHP 7 (mamba)
- Converted to XOOPS 2.5.8 Admin GUI (mamba)
- removed @version (mamba)
- PSR-2 code cosmetics (mamba)
- addNavigation(basename(__FILE__))  (mamba)
- dirname(__FILE__) to __DIR__ (mamba)
- replaced < br /> with < br > (mamba)
- Unnecessary double quotes (mamba)
- reference mismatch (mamba)
- Hardening to type safe (mamba)
- update Help (mamba)
- changed _handler to Handler (mamba)
- removed '/' from void HTML elements (&lt;br&gt;, &lt;img&gt;, etc) (mamba)
- HTML 5: removed 'selected' and 'checked' (mamba)
- updated file headers for consistency (mamba)
- renamed several $indexAdmin with $adminObject for consistency (mamba)
- Updates for PayPal changes (zyspec)
- added Contributing.md (mamba)
- moved functions to Utility class (mamba)
- include_once to require_once (mamba)
- php-cs-fixer updates (mamba)
- started conversion to XMF (mamba)
    &nbsp;&nbsp;- updated addInfoBoxLine entries (mamba)
    &nbsp;&nbsp;- displayNavigation(basename(__FILE__))  (mamba)
    &nbsp;&nbsp;- changed to ->displayButton('left') (mamba)

<h5>1.10 Beta 2       [2014-04-23]</h5>
<hr>

- moved all images, CSS, and JS files to /assets (mamba)
- renamed .html Smarty templates to .tpl (mamba)


<h5>1.10 Beta 1  [2013-03-23]</h5>
<hr>

- Converted to XOOPS 2.5.5 Admin GUI (Mamba)


<h5>1.00 Beta 1  [2006-11-02]</h5>
<hr>

- First public Release (Marcan)


xoops_version.php
<?php

/**
 *
 * Module: SmartClone
 * Author: The SmartFactory <www.smartfactory.ca>
 * Licence: GNU
 */

// defined('XOOPS_ROOT_PATH') || exit('XOOPS root path not defined');

$modversion['version']       = 1.10;
$modversion['module_status'] = 'Beta 3';
$modversion['release_date']  = '2017/07/20';
$modversion['name']          = 'SmartClone';
$modversion['author']        = 'The SmartFactory';
$modversion['description']   = _MI_SCLONE_MODULE_DESC;
$modversion['credits']       = 'Sudhaker, Solo';
$modversion['help']          = 'page=help';
$modversion['license']       = 'GNU GPL 2.0 or later';
$modversion['license_url']   = 'www.gnu.org/licenses/gpl-2.0.html';
$modversion['official']      = 0; //1 indicates supported by XOOPS Dev Team, 0 means 3rd party supported
$modversion['image']         = 'assets/images/logoModule.png';
$modversion['dirname']       = basename(__DIR__);
//$modversion['dirmoduleadmin']      = '/Frameworks/moduleclasses/moduleadmin';
//$modversion['icons16']             = '../../Frameworks/moduleclasses/icons/16';
//$modversion['icons32']             = '../../Frameworks/moduleclasses/icons/32';
$modversion['modicons16'] = 'assets/images/icons/16';
$modversion['modicons32'] = 'assets/images/icons/32';

$modversion['release_file']        = XOOPS_URL . '/modules/' . $modversion['dirname'] . '/docs/changelog.txt';
$modversion['module_website_url']  = 'www.xoops.org';
$modversion['module_website_name'] = 'XOOPS';
$modversion['min_php']             = '5.5';
$modversion['min_xoops']           = '2.5.9';
$modversion['min_admin']           = '1.2';
$modversion['min_db']              = array('mysql' => '5.5');

// Added by marcan for the About page in admin section
$modversion['developer_lead']         = 'marcan [Marc-André Lanciault]';
$modversion['developer_contributor']  = 'Sudhaker, Solo';
$modversion['developer_website_url']  = 'http://smartfactory.ca';
$modversion['developer_website_name'] = 'The SmartFactory';
$modversion['developer_email']        = 'marcan@smartfactory.ca';
$modversion['status_version']         = 'Beta';
$modversion['status']                 = 'Beta';
$modversion['date']                   = 'Unreleased';

$modversion['warning'] = _MI_SCLONE_WARNING_BETA;

$modversion['demo_site_url']     = '';
$modversion['demo_site_name']    = '';
$modversion['support_site_url']  = '';
$modversion['support_site_name'] = '';
$modversion['submit_bug']        = '';
$modversion['submit_feature']    = '';

$modversion['author_word'] = "This module was made possible by 2 great developers. The original cloning script was developed by Sudhaker for the SmartSection module. It was then improved by Solo who encapsulated Sudhaker's code into an easy web form interface for his Edito module. It was only after Solo have shown me his latest Edito that I had the idea to create a module that could clone any module in a '1 click operation'. Many thanks to Sudhaker and Solo for they made this module possible!";

// Admin things
$modversion['hasAdmin']    = 1;
$modversion['system_menu'] = 1;
$modversion['adminindex']  = 'admin/index.php';
$modversion['adminmenu']   = 'admin/menu.php';

// Search
$modversion['hasSearch'] = 0;

// Menu
$modversion['hasMain'] = 0;

// Templates
$modversion['templates'][1]['file']        = 'smartclone_index.tpl';
$modversion['templates'][1]['description'] = 'Help main page';

$modversion['templates'][2]['file']        = 'smartclone_register.tpl';
$modversion['templates'][2]['description'] = 'Page to inform user that he needs to be a member to use Live Help';

// Config options
$i = 0;
/*
 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_intro_message';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INTRO_MSG';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INTRO_MSG_DSC';
 $modversion['config'][$i]['formtype'] = 'textarea';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_register_title';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_REGISTER_TITLE';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_REGISTER_TITLE_DSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_register_image';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_REGISTER_IMAGE';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_REGISTER_IMAGE_DSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_register_text';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_REGISTER_TEXT';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_REGISTER_TEXT_DSC';
 $modversion['config'][$i]['formtype'] = 'textarea';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_register_footer';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_REGISTER_FOOTER';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_REGISTER_FOOTER_DSC';
 $modversion['config'][$i]['formtype'] = 'textarea';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_block_header';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_BLOCK_HEADER';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_BLOCK_HEADER_DSC';
 $modversion['config'][$i]['formtype'] = 'textarea';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'smartclone_block_footer';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_BLOCK_FOOTER';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_BLOCK_FOOTER_DSC';
 $modversion['config'][$i]['formtype'] = 'textarea';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_url';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_URL';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_URLDSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = 'http://www.inquiero.com';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_login';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_LOGIN';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_LOGINDSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_bgcolor';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_BGCOLOR';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_BGCOLORDSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_txtcolor';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_TXTCOLOR';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_TXTCOLORDSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_button';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_BUTTON';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_BUTTONSC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'inquiero_mode';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_MODE';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_MODESC';
 $modversion['config'][$i]['formtype'] = 'textbox';
 $modversion['config'][$i]['valuetype'] = 'text';
 $modversion['config'][$i]['default'] = '';

 ++$i;
 $modversion['config'][$i]['name'] = 'allow_anonymous';
 $modversion['config'][$i]['title'] = '_MI_SCLONE_INQUIERO_ANONYMOUS';
 $modversion['config'][$i]['description'] = '_MI_SCLONE_INQUIERO_ANONYMOUSDSC';
 $modversion['config'][$i]['formtype'] = 'yesno';
 $modversion['config'][$i]['valuetype'] = 'int';
 $modversion['config'][$i]['default'] = true;

 */
